server/api/archive.go

```diff
@@ -204,51 +204,7 @@ func (a *API) handleArchiveExportTeam(w http.ResponseWriter, r *http.Request) {
 	//     description: internal error
 	//     schema:
 	//       "$ref": "#/definitions/ErrorResponse"
-	if a.MattermostAuth {
-		a.errorResponse(w, r, model.NewErrNotImplemented("not permitted in plugin mode"))
-		return
-	}
-
-	vars := mux.Vars(r)
-	teamID := vars["teamID"]
-
-	ctx := r.Context()
-	session, _ := ctx.Value(sessionContextKey).(*model.Session)
-	userID := session.UserID
-
-	auditRec := a.makeAuditRecord(r, "archiveExportTeam", audit.Fail)
-	defer a.audit.LogRecord(audit.LevelRead, auditRec)
-	auditRec.AddMeta("TeamID", teamID)
-
-	isGuest, err := a.userIsGuest(userID)
-	if err != nil {
-		a.errorResponse(w, r, err)
-		return
-	}
-
-	boards, err := a.app.GetBoardsForUserAndTeam(userID, teamID, !isGuest)
-	if err != nil {
-		a.errorResponse(w, r, err)
-		return
-	}
-	ids := []string{}
-	for _, board := range boards {
-		ids = append(ids, board.ID)
-	}
-
-	opts := model.ExportArchiveOptions{
-		TeamID:   teamID,
-		BoardIDs: ids,
-	}
-
-	filename := fmt.Sprintf("archive-%s%s", time.Now().Format("2006-01-02"), archiveExtension)
-	w.Header().Set("Content-Type", "application/octet-stream")
-	w.Header().Set("Content-Disposition", "attachment; filename="+filename)
-	w.Header().Set("Content-Transfer-Encoding", "binary")
+	a.errorResponse(w, r, model.NewErrNotImplemented("not permitted in plugin mode"))
```

Comment:
@sbishel, as we only return an error response from this function after removing the `MattermostAuth` flag. Will this be a good idea to remove the full API end-point? Thoughts?